### PR TITLE
Add initial README.md for node-problem-detector.

### LINKED ISSUE #4: Write Readme.md for NodeProblemDetector
Write Readme.md for NodeProblemDetector to demonstrate:
* Motivation and scope of NodeProblemDetector
* Usage of current NodeProblemDetector
* Future plan of NodeProblemDetector.

/cc @kubernetes/node-problem-detector-maintainers

### PR DESCRIPTION
**Pull request by Random-Liu**
Fix #4.

This PR updated README.md of node-problem-detector.

@dchen1107 
/cc @kubernetes/sig-node

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,102 @@
...
+node-problem-detector aims to make various node problems visible to Kubernetes 
+control-plane. It is a [DaemonSet](http://kubernetes.io/docs/admin/daemons/)
+detecting node problems and reporting them to apiserver. Now it is running as
+a [Kubernetes Addon](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons)
+enabled by default in the GCE cluster.
+
+# Background
+There are tons of node problems could possibly affect the pods running on the
+node such as:
+* Hardware Issues: Bad cpu, memory or disk;
```
Comment (dchen1107) [inline README.md]: s/Issues/issues?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,102 @@
...
+* Kernel Issues: Kernel deadlock, corrupted file system;
```
Comment (dchen1107) [inline README.md]: ditto

File: README.md [hunk 3]
```diff
@@ -1,2 +1,102 @@
...
+* Container Runtime Issues: Unresponsive runtime daemon;
```
Comment (dchen1107) [inline README.md]: s/Runtime Issues/runtime issues

File: README.md [hunk 4]
```diff
@@ -1,2 +1,102 @@
...
+* ...
+
+Currently these problems are invisible to the control plane, so Kubernetes will
```
Comment (dchen1107) [inline README.md]: How about change this to "invisible to the upstream layers in cluster management stack"?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,102 @@
...
+continue scheduling pods to the bad nodes.
+
+To solve this problem, we introduced this new daemon **node-problem-detector** to
+detect node problems and make them visible to the control-plane. Once control-plane
```
Comment (dchen1107) [inline README.md]: This is a little bit misleading. The user might think a single node-problem-detector will detect all potential node problems, which is not the goal here. NodeProblemDetector is collecting the node problems from various daemons (only KernelMonitor for now, and rest of daemons are TBD).

File: README.md (hunk 1 above)
Comment (Random-Liu) [inline README.md]: ACK.

File: README.md (hunk 2 above)
Comment (Random-Liu) [inline README.md]: ACK

File: README.md (hunk 3 above)
Comment (Random-Liu) [inline README.md]: ACK

File: README.md (hunk 4 above)
Comment (Random-Liu) [inline README.md]: ACK. If so, I'll also update the design doc. :)

File: README.md (hunk 5 above)
Comment (Random-Liu) [inline README.md]: ACK. This is indeed not align with the goal. :)

**Comment by Random-Liu** [comment]
@dchen1107 Addressed comments.

**Comment by dchen1107** [comment]
LGTM

**Pull request merged**